compiler/rustc_lint/src/types.rs

```diff
@@ -727,7 +728,48 @@ declare_lint! {
     "proper use of libc types in foreign item definitions"
 }
 
-declare_lint_pass!(ImproperCTypesDefinitions => [IMPROPER_CTYPES_DEFINITIONS]);
+declare_lint! {
+    /// In it's platform C ABI, the AIX target follows the power alignment rule
+    /// for structs. This rule applies the natural alignment of the first struct
+    /// field if it is a floating-point type that is greater than 4-bytes, or
+    /// is an aggregate whose recursively "first" field is a floating-point
+    /// type greater than 4-bytes.
+    /// The alignment for subsequent fields of the associated structs use an
```

Comment:
So, is this also a correct description?

```suggestion
    /// In it's platform C ABI, AIX uses the "power" (as in PowerPC) alignment rule,
    /// which can also be set for XLC by `#pragma align power` or `-qalign=power`.
    /// Aggregates with a floating-point type as the recursively first field (as in "at offset 0")
    /// modify the layout of *subsequent* fields of the associated structs to use an
```

That is, given a hypothetical `#[repr(xlc_power)]` that would express this rule, and given these types:
```rust
#[repr(C, align(16))]
struct Aligned16 {
    _: bool,
}
#[repr(xlc_power)]
struct Whatever {
    float: f32,
    align16: Aligned16,
}
```

The following assert would still pass, right?
```rust
const _: () = assert!(align_of::<Whatever>() == 16);
```

So the only change is the second and later fields.

This way we don't have to describe "natural" alignment and we don't have to discuss the first field further, only its type.

Reply:
Yes, your thinking is correct. Thank you for the suggestion, I think it makes sense and it allows us to not have to mention the natural alignment explicitly.